quickcomm/views.py

```diff
@@ -69,5 +71,25 @@ def logout(request):
 
 
 def register(request):
-    # TODO implement register
-    return HttpResponse('Register Page')
+    if request.method == 'POST':
+        form = UserCreationForm(request.POST)
+        if form.is_valid():
+            user = form.save()
+            author = Author(user=user)
+            author.save()
+            # either log the user in or set their account to inactve
+            admin_approved = RegistrationSettings.objects.first().are_new_users_active
+            if admin_approved:
+                auth_login(request, user)
+            else:
+                user.is_active = False
+                user.save()
+            return redirect('/')
+        else:
+            form = UserCreationForm()
```

Comment:
Don't create a new form object here if the form is not valid. Instead, pass the original form back into the request.

The original form may contain data about what errors there were, so that way, the errors will show up on the form.

Just remove the two lines to fix this.